### PR TITLE
Modify `side_condition` results to be only `false` or `true` in Proof Hints

### PR DESCRIPTION
**Pull request by Robertorosmaninho**
This modification was proposed by the LLVM Backend and Math Proof Generation Teams from Pi2 to optimize the Hints generated by the `llvm-krun` as constructing and parsing the boolean values of the side conditions takes some time. Then, we choose to simplify the way we're generating these hints.

### CODE REVIEW

**Review by Baltoli — CHANGES_REQUESTED** [review]
Some questions and changes. Please also make sure that you accompany any PRs you open with a body that explains the context for the changes. This is important for reviewing _now_, as well as in the future when we look back at historical changes to try and understand them.

File: runtime/util/ConfigurationSerializer.cpp
Comment (Robertorosmaninho) [reply runtime/util/ConfigurationSerializer.cpp]: Good point! I've just done it!

File: lib/codegen/ProofEvent.cpp
Comment (Robertorosmaninho) [reply lib/codegen/ProofEvent.cpp]: Dwight pointed out to me that this is always a boolean: [see](https://github.com/runtimeverification/llvm-backend/blob/2c4eb2f0e80913a49c827a9821c51f9d133e7362/matching/src/main/scala/org/kframework/backend/llvm/matching/Matrix.scala#L773)!

File: include/kllvm/binary/ProofTraceParser.h
Comment (Robertorosmaninho) [reply include/kllvm/binary/ProofTraceParser.h]: Yes, that makes sense. I'm choosing to make `result` as a mandatory argument then.

File: include/kllvm/binary/ProofTraceParser.h
Comment (Robertorosmaninho) [reply include/kllvm/binary/ProofTraceParser.h]: Yes! Thanks for noticing that! I removed it everywhere here, but we'll need to modify `pyk` as well since I had to modify the bindings!

**Review by theo25 — APPROVED** [review]
This looks good to me but I think you should wait for Bruce to review your latest changes before you merge.

File: include/kllvm/binary/ProofTraceParser.h
Comment (Baltoli) [inline include/kllvm/binary/ProofTraceParser.h:125]: This is dead code now.
```suggestion
```

**Review by Baltoli — APPROVED** [review]
One minor comment, otherwise looks good!

File: include/kllvm/binary/ProofTraceParser.h
Comment (Robertorosmaninho) [reply include/kllvm/binary/ProofTraceParser.h]: Good point! Thank you!

**Pull request merged**
